pkg/telemetrylogs/json_query_builder.go

```diff
@@ -0,0 +1,660 @@
+	case qbtypes.FilterOperatorLessThan:
+		return sb.LT(fieldExpr, value), nil
+	case qbtypes.FilterOperatorLessThanOrEq:
+		return sb.LE(fieldExpr, value), nil
+	case qbtypes.FilterOperatorLike:
+		return sb.Like(fieldExpr, value), nil
+	case qbtypes.FilterOperatorNotLike:
+		return sb.NotLike(fieldExpr, value), nil
+	case qbtypes.FilterOperatorILike:
+		return sb.ILike(fieldExpr, value), nil
+	case qbtypes.FilterOperatorNotILike:
+		return sb.NotILike(fieldExpr, value), nil
+	case qbtypes.FilterOperatorRegexp:
+		return fmt.Sprintf("match(%s, %s)", fieldExpr, sb.Var(value)), nil
+	case qbtypes.FilterOperatorNotRegexp:
+		return fmt.Sprintf("NOT match(%s, %s)", fieldExpr, sb.Var(value)), nil
+	case qbtypes.FilterOperatorContains:
+		return sb.ILike(fieldExpr, fmt.Sprintf("%%%v%%", value)), nil
+	case qbtypes.FilterOperatorNotContains:
+		return sb.NotILike(fieldExpr, fmt.Sprintf("%%%v%%", value)), nil
+	case qbtypes.FilterOperatorIn, qbtypes.FilterOperatorNotIn:
+		// emulate IN/NOT IN using OR/AND over equals to leverage indexes consistently
+		values, ok := value.([]any)
+		if !ok {
+			values = []any{value}
+		}
+		conds := []string{}
+		for _, v := range values {
+			if operator == qbtypes.FilterOperatorIn {
+				conds = append(conds, sb.E(fieldExpr, v))
+			} else {
+				conds = append(conds, sb.NE(fieldExpr, v))
+			}
+		}
+		if operator == qbtypes.FilterOperatorIn {
+			return sb.Or(conds...), nil
+		}
+		return sb.And(conds...), nil
+	case qbtypes.FilterOperatorExists:
+		return fmt.Sprintf("%s IS NOT NULL", fieldExpr), nil
+	case qbtypes.FilterOperatorNotExists:
+		return fmt.Sprintf("%s IS NULL", fieldExpr), nil
+	default:
+		return "", qbtypes.ErrUnsupportedOperator
+	}
+}
+
+// GroupByArrayJoinInfo contains information about array joins needed for GroupBy
+type GroupByArrayJoinInfo struct {
+	ArrayJoinClauses []string // ARRAY JOIN clauses to add to FROM clause
+	TerminalExpr     string   // Terminal field expression for SELECT/GROUP BY
+}
+
+// BuildGroupBy builds GroupBy information for body JSON fields using arrayConcat pattern
+func (b *JSONQueryBuilder) BuildGroupBy(ctx context.Context, key *telemetrytypes.TelemetryFieldKey) (*GroupByArrayJoinInfo, error) {
+	path := strings.TrimPrefix(key.Name, BodyJSONStringSearchPrefix)
+
+	plan := PlanJSON(path, qbtypes.FilterOperatorExists, nil, b.IsPromoted(path), b.getTypeSet)
+	if len(plan) == 0 {
+		return nil, errors.Newf(errors.TypeInvalidInput, errors.CodeInvalidInput,
+			"Could not find any valid paths for: %s", path)
+	}
+
+	if plan[0].IsTerminal {
+		node := plan[0]
+
+		expr := fmt.Sprintf("dynamicElement(%s, '%s')", node.FieldPath(), node.TerminalConfig.ElemType.StringValue())
+		if b.IsPromoted(path) {
+			if len(plan) < 2 {
+				return nil, errors.Newf(errors.TypeUnexpected, CodePromotedPlanMissing,
+					"plan length is less than 2 for promoted path: %s", path)
+			}
+
+			// promoted column first then body_v2 column
+			expr = fmt.Sprintf("coalesce(%s, %s)",
```

Comment:
add a todo to change this once we have the new structure, which can be incorporated into the path table. This will be based on my changes of multi if